### ISSUE TITLE
PayPal fee not present in executed payment response

### ISSUE THREAD

**Issue by krico** [paypal/PayPal-Java-SDK#93]
I'm testing using the sdk against the sandbox and I don't get any fees (under Details) or anywhere else after executing a payment (Payment.execute).  

Is this a known bug?

**Comment by jziaja** [comment]
Hi @krico , the `amount.details.fees` property has been deprecated by the Payments API in favor of a newer [`transactionFee`](https://github.com/paypal/PayPal-Java-SDK/blob/master/rest-api-sdk/src/main/java/com/paypal/api/payments/Sale.java#L89) field that was recently added to all the SDKs.   You can find more information about this field in the [developer docs](https://developer.paypal.com/docs/api/#sale-object), as well as examples of how it will be returned.

I tested a payment out just now and was able to get the transaction fee in the response from execute call (using the .NET SDK):

![image](https://cloud.githubusercontent.com/assets/8227966/6667493/69223ae6-cbba-11e4-851f-ccbded67db52.png)

Can you verify whether or not you're seeing this new field?

**Comment by jziaja** [comment]
_cc [PayPal-Ruby-SDK Issue #14](https://github.com/paypal/PayPal-Ruby-SDK/issues/14)_

**Comment by jziaja** [comment]
Just to clarify, [version 1.2.0](https://github.com/paypal/PayPal-Java-SDK/releases/tag/v1.2.0) of this SDK includes the newer transaction fee properties.  If you're using an older version of the SDK, then you'll need to upgrade to version 1.2.0.

**Comment by krico** [comment]
@jziaja, I am using the latest and greatest java SDK (`com.paypal.sdk:rest-api-sdk:1.2.0`)

And I don't get the fees back neither under Details nor under transactionFee. 

In my setup I am also using a web-profile where I set the noShipping option to "1".  

See the executed Payment below (I printed this with `Payment.execute(...).toString()`

```
{
  "id": "PAY-3EH49608FN3060155KUDNXDQ",
  "create_time": "2015-03-16T13:33:02Z",
  "update_time": "2015-03-16T13:33:36Z",
  "intent": "sale",
  "payer": {
    "payment_method": "paypal",
    "payer_info": {
      "email": "jasify-ch@jasify.com",
      "first_name": "Christian",
      "last_name": "Asmussen",
      "payer_id": "D2DWHWYNBQ5T6"
    }
  },
  "transactions": [
    {
      "related_resources": [
        {
          "sale": {
            "id": "1LL289542W656834E",
            "create_time": "2015-03-16T13:33:02Z",
            "update_time": "2015-03-16T13:33:36Z",
            "amount": {
              "currency": "CHF",
              "total": "12.97"
            },
            "payment_mode": "INSTANT_TRANSFER",
            "state": "completed",
            "protection_eligibility": "INELIGIBLE",
            "parent_payment": "PAY-3EH49608FN3060155KUDNXDQ",
            "links": [
              {
                "href": "https://api.sandbox.paypal.com/v1/payments/sale/1LL289542W656834E",
                "rel": "self",
                "method": "GET"
              },
              {
                "href": "https://api.sandbox.paypal.com/v1/payments/sale/1LL289542W656834E/refund",
                "rel": "refund",
                "method": "POST"
              },
              {
                "href": "https://api.sandbox.paypal.com/v1/payments/payment/PAY-3EH49608FN3060155KUDNXDQ",
                "rel": "parent_payment",
                "method": "GET"
              }
            ]
          }
        }
      ],
      "amount": {
        "currency": "CHF",
        "total": "12.97",
        "details": {
          "shipping": "0.97",
          "subtotal": "12.00"
        }
      },
      "description": "Do I want this description?"
    }
  ],
  "state": "approved",
  "links": [
    {
      "href": "https://api.sandbox.paypal.com/v1/payments/payment/PAY-3EH49608FN3060155KUDNXDQ",
      "rel": "self",
      "method": "GET"
    }
  ]
}
```

**Comment by krico** [comment]
Ok, seems the problem was on my side, I upgraded the dependency on my pom.xml but never cleaned up the deployed webapp.  So I had both versions under WEB-INF/lib

```
target/app-1.0-SNAPSHOT/WEB-INF/lib/rest-api-sdk-1.1.2.jar
target/app-1.0-SNAPSHOT/WEB-INF/lib/rest-api-sdk-1.2.0.jar
```

I used the following code to detect this...
```java
            SDKVersion version = new SDKVersionImpl();
            log.warn("PayPal using SDK {} (version {})", version.getSDKId(), version.getSDKVersion());
```
Which unfortunately still prints
```
INFO: Running with PayPal SDK id=PayPal-Java-SDK (version 1.1.2)
```
I guess this is a totally different issue, but maybe bump-up the version in [SDKVersionImpl](https://github.com/paypal/PayPal-Java-SDK/blob/master/rest-api-sdk/src/main/java/com/paypal/base/sdk/info/SDKVersionImpl.java)

**Comment by juwlee** [comment]
Thanks @krico for the catch. It has been fixed in 8ed4ee3081d23b0340774e9f7e2aac060846e131
